### PR TITLE
Limit JDK versions for FIPS 140-2 matrix jobs

### PR DESCRIPTION
**Pull request by jkakavas**
We _only_ support Oracle JDK11 for FIPS 140-2 mode so there is no
need to run tests in FIPS 140-2 mode with all JDKs we otherwise
support. This change limits the versions for our matrix fips jobs
to JDK11 and openjdk16 which is the current bundled JDK. We want to
use the latter as a canary for changes that might break the
FIPS 140-2 mode, that might get introduced in JDK16 and later
backported to JDK11.

### CODE REVIEW

File: .ci/jobs.t/elastic+elasticsearch+periodic+java-fips-matrix.yml [hunk 1]
```diff
@@ -4,6 +4,6 @@ vars:
   - job-name: elastic+elasticsearch+%BRANCH%+periodic+java-fips-matrix
   - job-display-name: "elastic / elasticsearch # %BRANCH% - java fips compatibility matrix"
   - job-description: "Testing of the Elasticsearch %BRANCH% branch java FIPS compatibility matrix.\n"
-  - matrix-yaml-file: ".ci/matrix-runtime-javas.yml"
+  - matrix-yaml-file: ".ci/matrix-runtime-javas-fips.yml"
```
Comment (jkakavas) [inline .ci/jobs.t/elastic+elasticsearch+periodic+java-fips-matrix.yml:7]: Happy to get any feedback for tackling this in a different way. I'm not aware of how we could further filter the values from the existing matrix-runtime-javas.yml and keeping a specific file for these matrix jobs sounded like a good idea nevertheless. I understand we can define the job entirely here and add the runtime javas in an axis, let me know if this is a preferable approach

**Review by mark-vieira — APPROVED** [review]
LGTM. This is fine. In fact, it's exactly what I did for the 7.x branch to solve the same problem.

I'm wondering if we should keep jdk17 here though. I mean, this is going to be an LTS release so I suspect we should want to stay on top of issues there?

**Comment by jkakavas** [comment]
> I'm wondering if we should keep jdk17 here though. I mean, this is going to be an LTS release so I suspect we should want to stay on top of issues there

Good point. When we discussed about this, there weren't a jdk17 ea to consider so we decided on jdk16. I still think we need to keep jdk16 now and move to jdk17 on GA in September so that we don't get flaky tests from issues that will be resolved in GA of the LTS.

**Comment by mark-vieira** [comment]
Sure I think we should test on 17 EA if possible though just to stay on top of things. If the intent is to support 17 once it goes GA, we really should be testing now. Java 17 is already in rampdown phase so there shouldn't be any significant changes between now and then.

**Comment by jkakavas** [comment]
> Sure I think we should test on 17 EA if possible though just to stay on top of things. If the intent is to support 17 once it goes GA, we really should be testing now. Java 17 is already in rampdown phase so there shouldn't be any significant changes between now and then.

I'll talk with the rest of the team about replacing jdk16 with jdk17, we want to keep the matrix minimal so it will be either one. I will raise a follow up PR if need be

**Pull request merged**